src/redprl/tactic_elaborator.fun

```diff
@@ -368,12 +368,18 @@ struct
      | O.DEV_PRINT $ [_ \ tm'] =>
        (RedPrlLog.print RedPrlLog.INFO (getAnnotation tm, TermPrinter.ppTerm tm');
         T.idn)
+
+     | O.TAC_TRACE str $ _ => 
+       Lcf.rule o R.LabelGoal str
      | O.TAC_FAIL $ _ => fail "fail"
      | _ => raise RedPrlError.error [Fpp.text "Unrecognized tactic", TermPrinter.ppTerm tm]
 
   and multitactic_ sign env tm =
     case Tm.out tm of 
-       O.MTAC_ALL $ [_ \ tm] => T.all (tactic sign env tm)
+       O.MTAC_ALL $ [_ \ tm] =>
```

Comment:
this is sad; one day it won't be needed.

Reply:
I do not get it right a way. Could you possibly explain why it worked before?

Reply:
This is very ugly and unfortunate; what is happening is that the parser adds coercions between tactics and multitactics in various places (because we try not to expose to the user the difference). The parser tries to collapse these coercions if they are redundant (for instance, a roundtrip); but in this case, the roundtrip came into being after a substitution, so it had not been crushed.

The reason this results in a different behavior vis-a-vis tracing is that the trace is part of the global proof state and not the local state (which is just a judgment). So, coercing a multitactic to a tactic causes it to lose the global trace, and coercing it back to a multitactic effectively resets the trace.

The solution is to collapse this.